### ISSUE TITLE
Add new device codenames or pass input without checking against fixed list

### ISSUE THREAD

**Issue by ford--prefect** [matlink/gplaycli#277]
With gplaycli 3.29 on Python 3.9.2 on Archlinux I experience issues when passing 'flame' as device codename (-dc flag).
Installation and authentication method should not matter for that question.

This is most likely due to input checking against a fixed list of known codenames, any mismatch is rejected.
To prevent frequent updates of this list, one could simply pass the string on to google without checking against a fixed list.

Thank you for consideration, and for maintaining this very useful piece of software!

**Comment by OJFord** [comment]
Yes, the list is here:
https://github.com/matlink/googleplay-api/blob/master/gpapi/device.properties

Without digging into the code, (can't find a reference to it on a naive search) I assume it's not as straightforward as just 'passing the string on', but actually some ID/android version/something that's looked up from there and passed on based on codename. If that's the case, maybe that could be used directly instead of codename.

My device isn't listed, and they all seem to be old enough now that some apps don't support any of them, so it will at least need updating. I'd submit a PR if only knew how they were generated/from where the data was sourced.

**Comment by ford--prefect** [comment]
Good catch! This commit has a script, that seems to do the trick: https://github.com/matlink/googleplay-api/commit/13bc63b9279b6641756f97a83b9b3a5efef31bfc
It is located in scripts/fetch_devices.sh

**Comment by OJFord** [comment]
This seems to be the latest work:

https://gitlab.com/AuroraOSS/gplayapi/-/tree/master/src/main/resources

which seems to have sort of replaced 'Yalp Store', for which that original API was made, by the same author yeriomin.

I'm still unable to find a device that will allow me to download certain APKs though, even `org.mozilla.firefox` for example! (Which surely they make available for quite underpowered/old hardware.)

I've also tried to craft one specifically for my actual device (Nokia 3.4) using `adb shell getprop`, but I'm not having any luck there either.

**Comment by OJFord** [comment]
The Aurora store generates a configuration for the current device:
https://github.com/whyorean/AuroraStore/blob/a73afc22ec33f0442f8c7ef914c7ccb009da9459/app/src/main/java/com/aurora/store/data/providers/NativeDeviceInfoProvider.kt

Unfortunately it does so in Kotlin, and using the Android API...

As far as I can tell there's no way to view the result in the app either, that'd be nice.

**Comment by OJFord** [comment]
I built AuroraStore with a tiny patch to write out the properties:
https://gitlab.com/AuroraOSS/AuroraStore/-/merge_requests/117/diffs

(That was way easier than expected to build/test too - I don't do Android development but just installed Android Studio, pointed start up wizard at that existing project, and hit run, so should be easy to do for yourself too if that helps.)

However.. when I then ran `gplaycli` again (with the extracted properties as an extra profile in `gpapi`) I still had several 'your device is not compatible' errors. YMMV, but I can download & install the same apps with AuroraStore fine, so I'm thinking maybe it's a protobuf API change too.

For my own purposes (https://github.com/OJFord/terraform-provider-android) I'm now thinking I'll actually use AuroraStore as a sort of companion app to handle all this nastiness, and just send it an 'intent' (https://gitlab.com/AuroraOSS/AuroraStore/-/merge_requests/118/diffs) to download the apps I want, which I can then install with `adb` as before, (to avoid needing either root or manual dialogue confirmations) but without having to copy them over from `gplaycli`'s download dir.

**Comment by BrunoFischerGermany** [comment]
Hello!
is there an tutorial to add new devices zu the list? Maybe some new Samsung / XIAOMI / etc. Devices?

Thank you!
